### PR TITLE
[PoA] Persists polls repo in batches if need be

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs [hunk 1]
```diff
@@ -802,38 +802,59 @@ internal bool Synchronize(ChainedHeader newTip)
 
                 if (headers.Count > 0)
                 {
-                    this.PollsRepository.WithTransaction(transaction =>
+                    DBreeze.Transactions.Transaction currentTransaction = this.PollsRepository.GetTransaction();
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:805]: Perhaps add the `lock` back in

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs [hunk 2]
```diff
@@ -802,38 +802,59 @@ internal bool Synchronize(ChainedHeader newTip)
...
+
+                            this.logger.Info(progressString);
+                            this.signals.Publish(new FullNodeEvent() { Message = progressString, State = FullNodeState.Initializing.ToString() });
+
+                            this.PollsRepository.SaveCurrentTip(currentTransaction);
+
+                            currentTransaction.Commit();
+                            currentTransaction.Dispose();
+
+                            currentTransactionCommitted = true;
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:843]: Maybe just re-open the transaction instead of having this variable

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:843]: Done

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:805]: Done

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:843]: I did initially do this but then realized I had no way of knowing whether or not the transaction was committed when you exit the loop. `Dbreeze.Transaction` does not have a `IsCommitted` property or something similar.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs:843]: If you always reopen the transaction then you can always assume its not committed

**Review by quantumagi — APPROVED** [review]
LGTM

**Pull request merged**